src/issuance.rs

```diff
@@ -83,19 +82,26 @@ pub struct IssueInfo {
 }
 
 /// Compute the asset description hash for a given asset description.
-pub fn compute_asset_desc_hash(asset_desc: &[u8]) -> Result<[u8; 32], Error> {
-    if !is_asset_desc_of_valid_size(asset_desc) {
-        return Err(WrongAssetDescSize);
+///
+/// # Panics
+///
+/// Panics if `asset_desc` is not well-formed as per Unicode 15.0 specification, Section 3.9, D92.
+pub fn compute_asset_desc_hash(asset_desc: &NonEmpty<u8>) -> [u8; 32] {
+    // Check that asset_desc is well-formed as per Unicode 15.0 specification, Section 3.9, D92
+    let asset_desc_vec = asset_desc.iter().copied().collect::<Vec<u8>>();
+    if String::from_utf8(asset_desc_vec).is_err() {
+        panic!("asset_desc is not well-formed as per Unicode 15.0 specification, Section 3.9, D92");
```

Comment:
remove comment (`// Check ...`) repeating the same verbose function description and squeze into
```rust
    if String::from_utf8(asset_desc.iter().copied().collect::<Vec<u8>>()).is_err() {
        panic!("asset_desc is not a well-formed Unicode string");
    }
```

Reply:
(fix test error to match)

Reply:
Done